sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/non_streaming_order_by_aggregator.py

```diff
@@ -0,0 +1,152 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 Microsoft Corporation
+
+"""Internal class for multi execution context aggregator implementation in the Azure Cosmos database service.
+"""
+
+from azure.cosmos._execution_context.aio.base_execution_context import _QueryExecutionContextBase
+from azure.cosmos._execution_context.aio.multi_execution_aggregator import _MultiExecutionContextAggregator
+from azure.cosmos._execution_context.aio import document_producer
+from azure.cosmos._routing import routing_range
+from azure.cosmos import exceptions
+
+# pylint: disable=protected-access
+
+class _NonStreamingOrderByContextAggregator(_QueryExecutionContextBase):
+    """This class is a subclass of the query execution context base and serves for
+    non-streaming order by queries. It is very similar to the existing MultiExecutionContextAggregator,
+    but is needed since we're dealing with items and not document producers.
+
+    This class builds upon the multi-execution aggregator, building a document producer per partition
+    and draining their results entirely in order to create the result set relevant to the filters passed
+    by the user.
+    """
+
+    def __init__(self, client, resource_link, query, options, partitioned_query_ex_info):
+        super(_NonStreamingOrderByContextAggregator, self).__init__(client, options)
+
+        # use the routing provider in the client
+        self._routing_provider = client._routing_map_provider
+        self._client = client
+        self._resource_link = resource_link
+        self._query = query
+        self._partitioned_query_ex_info = partitioned_query_ex_info
+        self._sort_orders = partitioned_query_ex_info.get_order_by()
+        self._orderByPQ = _MultiExecutionContextAggregator.PriorityQueue()
+
+    async def __anext__(self):
+        """Returns the next result
+
+        :return: The next result.
+        :rtype: dict
+        :raises StopIteration: If no more result is left.
+        """
+        if self._orderByPQ.size() > 0:
+            res = await self._orderByPQ.pop_async(self._document_producer_comparator)
+            return res
+        raise StopAsyncIteration
+
+    async def fetch_next_block(self):
+
+        raise NotImplementedError("You should use pipeline's fetch_next_block.")
+
+    async def _repair_document_producer(self):
```

Comment:
Might be a dumb question but if this is used when there is a split, how does it use the Continuation from the previous producers in the children producers?

Reply:
it doesn't, it rebuilds the context from scratch

Reply:
What happens then if the query had already obtained some results? Would it get duplicates?

Reply:
any initial results would be discarded when the context gets rebuilt from scratch, so you won't have duplicates